jdaviz/app.py

```diff
@@ -254,12 +259,24 @@ def get_data_from_viewer(self, viewer_reference, data_label=None, cls=None):
         viewer = self.get_viewer(viewer_reference)
         cls = cls or viewer.default_class
 
-        data = [layer_state.layer.get_object(cls=cls)
-                if cls is not None else layer_state.layer
-                for layer_state in viewer.state.layers
-                if hasattr(layer_state, 'layer') and
-                isinstance(layer_state.layer, BaseData)
-                and (data_label is None or layer_state.layer.label == data_label)]
+        data = []
```

Comment:
I would advocate for returning a dict rather than list from this method, with keys set to the layer_state.layer.label values. That would make it easy to keep track of which Spectrum1D (or other object type getting returned) corresponds to which glue data in the context where this method was called. In my particular case that would let me populate the data selection dropdown from the dict keys and have those already indexed to the Spectrum1D objects.